data/parser/standard_array.go

```diff
@@ -182,7 +187,7 @@ func (s *StandardArray) ParseStringArray(index int) *[]string {
 		}
 
 		stringArrayValue = []string{}
-		parser, _ := NewStandardArray(s.context.NewChildContext(index), &arrayValue, IgnoreNotParsed)
+		parser, _ := NewStandardArray(s.context.NewChildContext(index), s.factory, &arrayValue, IgnoreNotParsed)
```

Comment:
The second return param is an `error` that only occurs if one of the first two arguments is `nil`. As we are already in a working StandardArray object that has already validated those two arguments, we know the call to NewStandardArray cannot fail and so we can ignore the error as it will always be nil.  This is a special case of an object having special internal knowledge of itself and can take certain shortcuts. If `NewStandardArray` was called in any other location then it would be good practice to check the return error.